pkgs/servers/moonraker/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, stdenvNoCC, fetchFromGitHub, python3, unstableGitUpdater }:
+
+let
+  pythonEnv = python3.withPackages (packages: with packages; [
+    tornado
+    pyserial
+    pillow
+    lmdb
+    streaming-form-data
+    distro
+    inotify-simple
+    libnacl
+    paho-mqtt
+  ]);
+in stdenvNoCC.mkDerivation rec {
+  pname = "moonraker";
+  version = "unstable-2021-07-18";
+
+  src = fetchFromGitHub {
+    owner = "Arksine";
+    repo = "moonraker";
+    rev = "42f61ceafa90fcfea8bffbe968e26a6fd8b61af6";
+    sha256 = "1w6l9pgs4n4nnk3h40y346bf6j3v4j4h1qnhj5dwlbwdxiqpd9gs";
+  };
+
+  installPhase = ''
+    mkdir -p $out $out/bin $out/lib
+    cp -r moonraker $out/lib
+
+    echo -e "#!/bin/sh\nexec ${pythonEnv}/bin/python $out/lib/moonraker/moonraker.py \"\$@\"" > $out/bin/moonraker
```

Comment:
Why don't you just use `patchShebangs`?

Reply:
Hmm, I'm not sure how to then add it to `bin` without using yet another wrapper. This seems simpler than doing two steps.